spec/requests/people_spec.rb

```diff
@@ -0,0 +1,201 @@
+ require 'rails_helper'
+
+RSpec.describe "/people", type: :request do
+  let(:valid_attributes) {
+    { name: "Pat McUser", phone: "212-555-1234", preferred_contact_method_id: create(:contact_method).id }.with_indifferent_access
+  }
+  let(:invalid_attributes) {
+    { name: nil }
+  }
+
+  describe "GET /index" do
+    subject { get people_url }
+
+    before(:each) { create(:person) } 
+
+    it_behaves_like "basic user authorization required" 
+  end
+
+  describe "GET /show" do
+    let(:resource) { create(:person) }
+    subject { get person_url(resource) }
+
+    it_behaves_like "basic user authorization required"
+  end
+
+  describe "GET /new" do
+    subject { get new_person_url }
+
+    it_behaves_like "basic user authorization required"
+  end
+
+  describe "GET /edit" do
+    context "when editing your own record" do
+      let(:resource) { create(:user, :with_person) }
+      subject { get edit_person_url(resource.person) }
+
+      it_behaves_like 'redirects without authorization'
+
+      # No easy way to use the shared example, so just do it inline
+      it "renders successfully when signed-in" do
+        sign_in(resource)
+        subject
+        expect(response).to be_successful
+      end
+    end
+
+    context "When editing another user's record" do
+      let(:resource) { create(:person) }
+      subject { get edit_person_url(resource) }
+
+      it_behaves_like "redirects without authorization"
+      it_behaves_like "disallowed behavior as", [:user, :with_person]
+      it_behaves_like "successful with admin user authorization"
+    end
+  end
+
+  describe "POST /create" do
+    subject { post people_url, params: { person: valid_attributes } }
+
+    it_behaves_like "basic user authorization required"
+
+    context "when the user already has a person record for themselves" do
+      let(:self) { create(:user) }
+      subject { post people_url, params: { person: valid_attributes.merge(user_id: self.id) } }
+
+      before(:each) { create(:person, user_id: self.id) }
+
+      # TODO: What happens when the user already has a record created for themselves
```

Comment:
🤷‍♀️

Reply:
haha 😆 

This could be decided on later, it was just something that came to mind while I was doing it